adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -849,6 +850,15 @@ private String checkInputParameters(final String resource, final String clientId
             redirectUri = inputRedirectUri;
         }
 
+        //check the permissions required for the broker usage
+        if (AuthenticationSettings.INSTANCE.getUseBroker()) {
+            try {
+                mBrokerProxy.verifyManifestPermissions();
+            } catch (final UsageAuthenticationException exception) {
+                callback.onError(exception);
```

Comment:
final String redirectUri should also go below this block

---
In reply to: [75787853](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/698#discussion_r75787853) [](ancestors = 75787853)